Foosball/Foosball/BallTracker.cs

```diff
@@ -185,10 +202,35 @@ private bool IsBallLost()
 
         private void UpdateScores()
         {
+            CheckForWinner();
             lbScoreA.Text = _scoreA.ToString();
             lbScoreB.Text = _scoreB.ToString();
         }
 
+        private void CheckForWinner()
+        {
+            if(_teamNames.Count() != 0)
+            {
+                if(_scoreA >= 10)
+                {
+                    _teamNames.Add(_teamNames[_round * 2 - 2]);
+                    _round++;
+                    this.Hide();
+                    var load = new TournamentBracket(_teamNames, _round);
+                    load.ShowDialog();
+                    this.Close();
+                }
+                if (_scoreB >= 10)
+                {
+                    _teamNames.Add(_teamNames[_round * 2 - 1]);
+                    _round++;
+                    this.Hide();
+                    var load = new TournamentBracket(_teamNames, _round);
+                    load.ShowDialog();
+                    this.Close();
```

Comment:
217-221 rows are the same as 226-230. You could refactor it in one method or one if. For example:
```
if (_scoreA >= 10 || _scoreB >= 10)
 {
	if (_scoreA >= 10)
	{		
                _teamNames.Add(_teamNames[_round * 2 - 2]);
	}
	else
	{
		_teamNames.Add(_teamNames[_round * 2 - 1]);
	}
	
	_round++;
	this.Hide();
	var load = new TournamentBracket(_teamNames, _round);
	load.ShowDialog();
	this.Close();
 }
```